cli/src/local/watcher.js

```diff
@@ -125,6 +137,26 @@ class LocalWatcher {
     })
   }
 
+  handleEvents (events: ChokidarFSEvent[]) {
+    const [unlinkFileEvents, nonUnlinkFileEvents] = _.partition(events, e => e.type === 'unlink')
+    _.each(unlinkFileEvents, e => { this.onUnlinkFile(e.path) })
+    const [nonUnlinkDirEvents, unlinkDirEvents] = _.partition(nonUnlinkFileEvents, e => e.type !== 'unlinkDir')
+    _.each(nonUnlinkDirEvents, e => {
+      switch (e.type) {
+        case 'add':
+          this.onAddFile(e.path, e.stats)
+          break
+        case 'addDir':
+          this.onAddDir(e.path, e.stats)
+          break
+        case 'change':
+          this.onChange(e.path, e.stats)
+          break
+      }
+    })
+    _.each(unlinkDirEvents, e => { this.onUnlinkDir(e.path) })
```

Comment:
@aenario Tests pass with this version, but I'm almost sure it's actually broken.
Here we naively change the order of the events, which may fix some event order issues but probably also breaks other cases...

Passing the events to the LocalWatcher methods as they come from the buffer would probably match the previous behavior more closely, but it breaks the tests.